File: binarySearch.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int low = 0;
        int high = nums.size() - 1;
        
        while(low <= high) {
            int mid = (low + high) / 2;
            
            if(target == nums[mid]) {
                return mid;
            }
            else if(target > nums[mid]) {
                low = mid + 1;
            }
            else {
                high = mid - 1;
            }
        }
        
        return -1;
    }
};

//Runtime: 63 ms, faster than 22.51% of C++ online submissions for Binary Search.
//Memory Usage: 27.5 MB, less than 92.51% of C++ online submissions for Binary Search.
//https://leetcode.com/problems/binary-search/submissions/

File: binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, r = nums.length - 1;
        
        while(l <= r) {
            int center = (l + r) / 2;
            if(nums[center] == target) return center;
            else if(nums[center] < target) l = center + 1;
            else r = center - 1;
        }
        
        return -1;
    }
}

// https://leetcode.com/problems/binary-search/submissions/
// Runtime: 0 ms, faster than 100.00% of Java online submissions for Binary Search.
// Memory Usage: 42.6 MB, less than 98.78% of Java online submissions for Binary Search.
